addons/pos_mrp/tests/test_pos_mrp_flow.py

```diff
@@ -332,3 +332,115 @@ def test_bom_kit_different_uom_invoice_valuation(self):
         self.assertEqual(expense_line.filtered(lambda l: l.product_id == self.kit).debit, 6000.0)
         interim_line = order.account_move.line_ids.filtered(lambda l: l.account_id.id == accounts['stock_output'].id)
         self.assertEqual(interim_line.filtered(lambda l: l.product_id == self.kit).credit, 6000.0)
+
+    def test_bom_kit_order_total_cost_with_shared_component(self):
+        category = self.env['product.category'].create({
+            'name': 'Category for average cost',
+            'property_cost_method': 'average',
+        })
+
+        self.kit_1 = self.env['product.product'].create({
```

Comment:
No need to link those records to `self`
```suggestion
        kit_1 = self.env['product.product'].create({
```
It's useful only when defined in the `setUpClass`, when you want to use them in multiple tests

Reply:
Thanks, I’ll change it. When I first started writing it, I was just copying and pasting other people’s tests, but I’ve realized that’s not necessary. Thank you for pointing it out.